dartagnan/src/test/java/com/dat3m/dartagnan/spirv/SpirvHeaderTest.java

```
@@ -0,0 +1,425 @@
+                ; @Output: not exists (%v2!=123 or %v3!=0)
+                ; @Config: 1, 1, 1
+                """,
+                        "forall:!%v1!=bv64(7) && !%v2!=bv64(123) && !%v3!=bv64(0)", new HashMap<>()},
+                {TestType.COMPLEX, "", "", new HashMap<>()},
+        });
+    }
+
+    @Test
+    public void testValidFormatHeader() {
+        Assume.assumeTrue(type == TestType.VALID_FORMAT);
+        try {
+            parse();
+        } catch (Exception e) {
+            fail(msg);
+        }
+    }
+
+    @Test
+    public void testInvalidFormatHeader() {
+        Assume.assumeTrue(type == TestType.INVALID_FORMAT);
+        try {
+            parse();
+        } catch (Exception e) {
+            assert (e.getMessage().equals(msg));
+        }
+    }
+
+    @Test
+    public void testValueHeader() {
+        Assume.assumeTrue(type == TestType.VALUE);
+        try {
+            Program program = parse();
+            for (Map.Entry<String, String> entry : values.entrySet()) {
+                Optional<MemoryObject> v = program.getMemory().getObjects().stream().filter(o -> o.getCVar().equals(entry.getKey())).findFirst();
+                if (entry.getValue().contains(",")) {
+                    String[] values = Arrays.stream(entry.getValue().split(","))
+                            .map(String::trim)
+                            .toArray(String[]::new);
+                    assert (v.isPresent());
+                    for (String s : values) {
+                        String[] offsetAndValue = s.split(":");
+                        assert v.get().getInitialValue(Integer.parseInt(offsetAndValue[0])).toString().equals(offsetAndValue[1].trim());
+                    }
+                } else {
+                    assert (v.isPresent());
+                    assert (v.get().getInitialValue(0).toString().equals(entry.getValue()));
+                }
+            }
+        } catch (Exception e) {
+            fail(e.getMessage());
```

We can drop try-catch and let the test fail itself if it fail